parser.go

```diff
@@ -98,6 +98,13 @@ func (p *Parser) parseExpression(stop ...rune) (Expression, error) {
 			continue
 		}
 
+		// Ignore empty variable references
+		if strings.HasPrefix(p.input[p.pos:], `$ `) {
+			p.pos += 2
+			expr = append(expr, ExpressionItem{Text: `$ `})
+			continue
+		}
```

Comment:
we might wanna check for common non-space whitespace too, the `\n`s and `\r`s of the world